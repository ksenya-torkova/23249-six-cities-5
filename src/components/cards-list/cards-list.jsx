import {cardListTypes} from "../../prop-types";
import CitiesPlaceCard from "../cities-place-card/cities-place-card";
import React from "react";

const CardList = (props) => {
  const {rooms} = props;

  return (
    <div className="cities__places-list places__list tabs__content">
      {rooms.map((room) => (
        <CitiesPlaceCard
          cardData = {room}
          key = {room.id}
        />
      ))}
    </div>
  );
};

CardList.propTypes = cardListTypes;

export {CardList};
export default React.memo(CardList);
