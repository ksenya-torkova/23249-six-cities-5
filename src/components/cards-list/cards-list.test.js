import {CardList} from "./cards-list";
import React from "react";
import renderer from "react-test-renderer";

it(`Should CardList render properly`, () => {
  const tree = renderer
    .create(
        <CardList
          rooms={[]}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
