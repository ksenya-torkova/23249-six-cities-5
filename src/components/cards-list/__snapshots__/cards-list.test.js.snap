// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CardList render properly 1`] = `
<div
  className="cities__places-list places__list tabs__content"
/>
`;
